CONTRIBUTING.md

```diff
@@ -154,20 +195,37 @@ maintenance branche(s).
 If this is the case, the fix should be implemented against the earliest relevant
 maintenance branch, which will be indicated in the issue as the _milestone_.
 
-For instance, if an issue should be fixed in both 3.1 and 3.2, the later being
-developed on main, the fix PR should be opened against `3.1.x` (and the issue
-will be triaged into `3.1.x Maintenance Backlog`).
+For instance, if an issue should be fixed in both 3.3 and 3.4, the later being
+developed on main, the fix PR should be opened against `3.3.x` (and the issue
+will be triaged into `3.3.x Backlog`).
 
 If on the other hand the issue should only be fixed in the next feature release,
-it will be affected to the `Backlog` triage milestone and should be developed
-against `main`.
+assuming `3.4.x` is the current main line it will be affected to the `3.4.x Backlog`
+triage milestone and should be developed against `main`.
+
+### From PR to `main`
+For each PR, there will generally be **a single commit** that goes into `main` (or formerly `master`). We will try to use the `squash-and-merge` strategy on GitHub to that effect, but if you do that yourself once the PR is approved, it's even better :heart:
```

Comment:
```suggestion
For each PR, there will generally be **a single commit** that goes into `main` (or formerly `master`). We will try to use the `squash-and-merge` strategy on GitHub to that effect.

Reply:
We discussed that we want to preserve the history so it is better that the `squash-and-merge` is done only by maintainers via the GitHub UI

Reply:
I'm not sure if you're referring to preserving the history of production branches or to the fact that force-pushes in PRs makes reviewing harder for us? in any case I'm ok with removing that sentence: if we want the contributor to do some polishing of their PR branch before merging, we can always ask them on a case by case basis.

Reply:
yeah we wanted to not have force pushes if that's possible